app/services/summary_service.rb

```diff
@@ -46,6 +46,7 @@ def self.summary(summary_type, prison, page, params)
     end
 
     if params.sort_field.present?
+      bucket.items.each { |offender| add_reception_date(offender) } unless summary_type == :allocated
```

Comment:
If this is sorting before only taking a page worth of offenders for the view, it might well trigger hundreds of api calls via add_reception_date :(

Reply:
agreed 100%

Reply:
yes - that's why the call it is making is cached. It takes around 30 seconds uncached, but is almost instantaneous when cached.